### PR TITLE
GCS all types e2e tests

### PR DESCRIPTION
**Pull request by bharatgulati**

### CODE REVIEW

File: src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature [hunk 1]
```diff
@@ -0,0 +1,87 @@
+@GCS_Source
+Feature: GCS source - Verification of GCS to GCS Additional Tests successful
+
+  @GCS_AVRO_FILE @GCS_SINK_TEST
+  Scenario: To verify data is getting transferred from GCS to GCS using Avro and Json file format
```
Comment (itsmekumari) [inline src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature]: mention the scenario as its datatypes related as well

File: src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    Then Close the pipeline logs
+    Then Click on preview data for GCS sink
+    Then Verify preview output schema matches the outputSchema captured in properties
+    Then Close the preview data
+    Then Deploy the pipeline
+    Then Run the Pipeline in Runtime
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Validate the data from GCS to GCS with Expected avro file and target bucket
```
Comment (itsmekumari) [inline src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature]: Change the step to be more clear

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java [hunk 3]
```diff
@@ -920,4 +920,8 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
     PluginPropertyUtils.addPluginProp(" bqTargetTable",  bqTargetTable);
     BeforeActions.scenario.write("BQ Target Table " +  bqTargetTable + " updated successfully");
   }
+  @Before(order = 1, value = "@GCS_AVRO_FILE")
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:923]: Add the after hook also for this

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+   * @param bucketName The name of the GCS bucket to validate.
+   * @return True if the GCS data matches the Avro data, false otherwise.
+   * @throws IOException If an IO error occurs during data retrieval.
+   */
+  public static boolean validateGCSSourceToGCSSink(String bucketName) throws IOException {
+    Map<String, JsonObject> expectedAvroData = convertAvroToJsonWithKeys();
+    Map<String, JsonObject> actualGcsData = listBucketObjects(bucketName);
+
+    boolean isMatched = actualGcsData.equals(expectedAvroData);
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: nit: remove blank line

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+   * obtained from Avro files converted to JSON.
+   *
+   * @param bucketName The name of the GCS bucket to validate.
+   * @return True if the GCS data matches the Avro data, false otherwise.
+   * @throws IOException If an IO error occurs during data retrieval.
+   */
+  public static boolean validateGCSSourceToGCSSink(String bucketName) throws IOException {
+    Map<String, JsonObject> expectedAvroData = convertAvroToJsonWithKeys();
+    Map<String, JsonObject> actualGcsData = listBucketObjects(bucketName);
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: nit: remove blank line

File: src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    List<String> bucketObjectNames = objectNamesWithData.collect(Collectors.toList());
+
+    if (!bucketObjectNames.isEmpty()) {
+      String objectName = bucketObjectNames.get(0);
+      if (objectName.contains("part-r")) {
+        Map<String, JsonObject> dataMap2 = fetchObjectData(projectId, bucketName, objectName);
+        bucketObjectData.putAll(dataMap2);
+      }
+    }
+
```
Comment (itsmekumari) [inline src/e2e-test/java/io/cdap/plugin/gcs/GCSValidationHelper.java]: nit: remove blank line, please check in file.

File: src/e2e-test/resources/pluginParameters.properties [hunk 7]
```diff
@@ -128,6 +128,11 @@ gcsReadRecursivePathSchema=[{"key":"id","value":"int"},{"key":"name","value":"st
   {"key":"yearofbirth","value":"int"},{"key":"tablename","value":"string"},{"key":"offset","value":"string"}]
 gcsParquetFile=testdata/GCS_PARQUET_TEST.parquet
 gcsAvroFile=testdata/GCS_AVRO_TEST.avro
+gcsAvroAllTypeData=[{"key":"id","value":"long"},{"key":"array","value":"string"},\
```
Comment (itsmekumari) [inline src/e2e-test/resources/pluginParameters.properties]: Add schema text at the end of parameter.

Comment (bharatgulati) [reply src/e2e-test/resources/pluginParameters.properties]: Done

File: src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java (hunk 3 above)
Comment (bharatgulati) [reply src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java:923]: Added

File: src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature (hunk 2 above)
Comment (bharatgulati) [reply src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature]: Changed the step.

File: src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature (hunk 1 above)
Comment (bharatgulati) [reply src/e2e-test/features/gcs/source/GCSToGCSAdditonalTests.feature]: Done.

**Review by itsmekumari — APPROVED** [review]

**Pull request merged**